clang/lib/CodeGen/CGCUDANV.cpp

```diff
@@ -748,8 +748,12 @@ llvm::Function *CGNVCUDARuntime::makeModuleCtorFunction() {
   llvm::Constant *FatBinStr;
   unsigned FatMagic;
   if (IsHIP) {
-    FatbinConstantName = ".hip_fatbin";
-    FatbinSectionName = ".hipFatBinSegment";
+    FatbinConstantName = CGM.getTriple().isMacOSX()
+                               ? "__HIP,__hip_fatbin"
+                               : ".hip_fatbin";
+    FatbinSectionName = CGM.getTriple().isMacOSX()
+                               ? "__HIP,__fatbin"
+                               : ".hipFatBinSegment";
```

Comment:
These changes are mimicking the existing CUDA ternary operators / names. However, I'm not sure if this works already and if the names are good.
I'm not really sure how segments / sections work on macOS (although I'm quite familar with ELF and COFF).